src/test/functional/custom-steps.ts

```diff
@@ -9,7 +9,7 @@ export = function () {
       this.fillField('#username', username);
       this.fillField('#password', password);
       this.wait(5);
-      this.click('Sign in');
+      tryTo(() => this.click('Sign in'));
```

Comment:
wouldn't this pass even if the test failed to click "sign in"? Feels like that's going to confuse someone when it breaks a test further down the line

Reply:
in idam-web-public we use "waitForText" and "waitInUrl" to help ensure the page is ready - have we tried that?

Reply:
It doesn't need to fail if it's succeeded by another assertion. As this in the login custom function, that's invariably the case.

Reply:
Not sure what you mean - this would be in another test as "loginAs" and when that finishes you would expect to be logged in. Are you saying the other assertions would be in the test that now it assumes it's logged in?

Reply:
No, I'm saying that if login failed, subsequent assertions would fail.

Reply:
yeah - and I'm saying that's confusing. I'm not sure that "tryTo" is the right solution, unless you were going to have an assertion based on the result of the tryTo

Reply:
I'm afraid we're going to have to disagree on this, then.

Reply:
@pandaSheltie you mentioned this helped with the reliability of the tests - do we understand why?

Reply:
`this.click('Sign in')` was a recurring point of failure because Codecept uses default assertions on all its action functions. So tests were failing even though the click, as far as the browser was concerned, had succeeded and loaded the next page. Codecept, however, somehow can get out of sync with the browser (maybe because we're running eight tests in parallel, I don't know). It looks for a button with the label "Sign in" and can't find it, so the test fails. Using `tryTo` means the click doesn't fail automatically.

Reply:
That all makes sense - however, I'd expect it to fail on the line before too `this.fillField('#password', password);`. Why would it not be failing there?

Reply:
They're not failing because the form hasn't been submitted at that point. It's like there's two threads, one initiating the actions the and the other asserting on the state of elements in the page (not saying this is what is actually happening, although it does hint at some kind of race condition).

Reply:
We've agreed to add an assertion that the user is indeed successfully logged in before exiting the `loginAs` method.

Reply:
As this method is not always used to assert successful logins, the above suggestion does not work.
We've renamed the method to "tryLoginAs" to make it less confusing.